wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -337,6 +337,20 @@ public TypeSpec generateEnum(EnumType type) {
 
   /** Returns the generated code for {@code type}, which may be a top-level or a nested type. */
   public TypeSpec generateMessage(MessageType type) {
+    NameAllocator nameAllocator = new NameAllocator();
+    nameAllocator.newName("serialVersionUID", "serialVersionUID");
+    nameAllocator.newName("ADAPTER", "ADAPTER");
+    nameAllocator.newName("MESSAGE_OPTIONS", "MESSAGE_OPTIONS");
+    nameAllocator.newName("tagMap", "tagMap");
+    nameAllocator.newName("hashCode", "hashCode");
```

Comment:
Still though, we could always `super.hashCode` qualify and do we care if `tagMap` gets renamed? I think users would prefer their message field name retained `tagMap` instead of some argument in the ctor they'll never use.